src/test/ui/resolve/missing-std.rs

```diff
@@ -0,0 +1,8 @@
+// compile-flags: --target x86_64-unknown-uefi
+// rustc-env:CARGO=/usr/bin/cargo
+// rustc-env:RUSTUP_HOME=/home/bors/.rustup
+// error-pattern: can't find crate for `std`
+// error-pattern: target may not be installed
+// error-pattern: consider building the standard library from source with `cargo build -Zbuild-std`
+// error-pattern: consider downloading the target with `rustup target add x86_64-unknown-uefi`
```

Comment:
Test output is predictable here, so it should be able to use regular `//~ ERROR` annotations instead of `// error-pattern`?

Also, could you move the test from `resolve` to `crate-loading`?

Reply:
> it should be able to use regular //~ ERROR annotations instead of // error-pattern?

I used error-pattern because the error didn't have a span. But I think now that I added `#![no_std]` and `extern crate std` it should have a span and I can use `ERROR` annotations.

That also made me notice that right now I suggest using `#![no_std]` unconditionally; I switched it to only suggesting no_std if this didn't come from `extern crate std`.

> Also, could you move the test from resolve to crate-loading?

Done.

Reply:
> That also made me notice that right now I suggest using #![no_std] unconditionally; I switched it to only suggesting no_std if this didn't come from extern crate std.

I'm not sure how to test this (because it requires building core but not std), but I tested locally with `-Zbuild-std=core` and it appears to work correctly:

```
$ cargo +stage1 build --target x86_64-unknown-uefi -Zbuild-std=core -q
error[E0463]: can't find crate for `std`
 --> src/main.rs:2:1
  |
2 | extern crate std;
  | ^^^^^^^^^^^^^^^^^ can't find crate
  |
  = note: the `x86_64-unknown-uefi` target may not support the standard library
  = help: consider building the standard library from source with `cargo build -Zbuild-std`
  = help: consider downloading the target with `rustup target add x86_64-unknown-uefi`
```